wrapanapi/systems/openstack.py

```diff
@@ -130,17 +132,26 @@ def _get_state(self):
 
     def _get_networks(self):
         self.refresh()
-        # TODO: Do we really need to access a private attr here?
-        return self.raw._info['addresses']
+        return self.raw.addresses
 
-    @property
-    def ip(self):
+    def get_ip_by_type(self, ip_type):
         networks = self._get_networks()
         for network_nics in networks.values():
             for nic in network_nics:
-                if nic['OS-EXT-IPS:type'] == 'floating':
+                if nic['OS-EXT-IPS:type'] == ip_type:
                     return str(nic['addr'])
 
+    @property
```

Comment:
Please see #386 for updates to IP lookup.

Reply:
Thanks, will rebase than and test it.

Reply:
looked at the code in that PR, think for openstack nothing should change for this PR, as all_ips will return all the available netwoks ips, but here we need external ip, especially when assiging a floating ip, all_ips will return internal and external (as I guess) [ip property should be external ip (fixed or floating)].